packages/databricks-vscode/src/bundle/BundleProjectManager.ts

```diff
@@ -0,0 +1,143 @@
+import {
+    QuickPickItem,
+    QuickPickItemKind,
+    Disposable,
+    Uri,
+    window,
+    commands,
+} from "vscode";
+import {ConnectionManager} from "../configuration/ConnectionManager";
+import {ConfigModel} from "../configuration/models/ConfigModel";
+import {BundleFileSet} from "./BundleFileSet";
+import {logging} from "@databricks/databricks-sdk";
+import {Loggers} from "../logger";
+import {CachedValue} from "../locking/CachedValue";
+import {CustomWhenContext} from "../vscode-objs/CustomWhenContext";
+
+export class BundleProjectManager {
+    private logger = logging.NamedLogger.getOrCreate(Loggers.Extension);
+    private disposables: Disposable[] = [];
+
+    private _isBundleProject = new CachedValue<boolean>(async () => {
+        const rootBundleFile = await this.bundleFileSet.getRootFile();
+        return rootBundleFile !== undefined;
+    });
+
+    public onDidChangeStatus = this._isBundleProject.onDidChange;
```

Comment:
Let's not expose this. This class is too meta for other classes to take a dependency on this. I would want it to drive all downstream classes (as it is doing now), but nothing else should take a dependency.